apps/emqx/src/emqx_persistent_session_ds_router.erl

```diff
@@ -0,0 +1,230 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_persistent_session_ds_router).
+
+-include("emqx.hrl").
+-include("emqx_persistent_session_ds/emqx_ps_ds_int.hrl").
+
+-export([init_tables/0]).
+
+%% Route APIs
+-export([
+    do_add_route/2,
+    do_delete_route/2,
+    match_routes/1,
+    lookup_routes/1,
+    foldr_routes/2,
+    foldl_routes/2
+]).
+
+-export([cleanup_routes/1]).
+-export([print_routes/1]).
+-export([topics/0]).
+
+-ifdef(TEST).
+-export([has_route/2]).
+-endif.
+
+-type dest() :: emqx_ds:session_id().
+
+-export_type([dest/0]).
+
+%%--------------------------------------------------------------------
+%% Table Initialization
+%%--------------------------------------------------------------------
+
+init_tables() ->
+    mria_config:set_dirty_shard(?PS_ROUTER_SHARD, true),
+    ok = mria:create_table(?PS_ROUTER_TAB, [
+        {type, bag},
+        {rlog_shard, ?PS_ROUTER_SHARD},
+        {storage, ram_copies},
```

Comment:
Q: Shouldn't we preserve them on disk as well?

Reply:
True, I took what we have for the normal router, but it probably makes more sense to keep this in disk.  Perhaps we should set it to `rocksdb_copies` (or fallback to `disc_copies` for Windows).